packages/core/components/grouper/index.js

```diff
@@ -2,128 +2,57 @@ import React from "react";
+  width: 100%;
+  height: 100%;
 
-  > *:focus {
-    z-index: 1;
-  }
+  ${props => {
+    return props.gutter !== undefined
+      ? css`margin: calc(-0.5 * ${themeGet(`space.${props.gutter}`, props.gutter)});`
+      : css`margin: calc(-0.5 * ${themeGet("grouper.space.gutter")});`
+  }}
 
   > * {
-    ${props => props.leadingMargin}: ${props =>
-      themeGet(`space.${props.gutter}`, props.gutter)};
+    ${props => {
+      return props.gutter !== undefined
+        ? css`margin: calc(0.5 * ${themeGet(`space.${props.gutter}`, props.gutter)});`
+        : css`margin: calc(0.5 * ${themeGet("grouper.space.gutter")});`
+    }}
   }
-`;
 
-const CompactGrouper = styled(StyledGrouper).attrs(props => ({
-  leadingBorder:
-    props.flexDirection === "column" ? "border-top" : "border-left",
-  firstChildCorner:
-    props.flexDirection === "column"
-      ? "border-top-right-radius"
-      : "border-bottom-left-radius",
-  lastChildCorner:
-    props.flexDirection === "column"
-      ? "border-bottom-left-radius"
-      : "border-top-right-radius"
-}))`
-  > * {
-    border-radius: 0 !important;
-  }
-
-  ${props =>
-    props.gutter === 0 &&
-    css`
-      > * + * {
-        ${props => props.leadingBorder}: 0;
-      }
-    `}
-
-  > :first-child {
-    border-top-left-radius: ${props =>
-      themeGet(`radii.${props.borderRadius}`, props.borderRadius)} !important;
-    ${props => props.firstChildCorner}: ${props =>
-      themeGet(`radii.${props.borderRadius}`, props.borderRadius)} !important;
-  }
-
-  > :last-child {
-    ${props => props.lastChildCorner}: ${props =>
-      themeGet(`radii.${props.borderRadius}`, props.borderRadius)} !important;
-    border-bottom-right-radius: ${props =>
-      themeGet(`radii.${props.borderRadius}`, props.borderRadius)} !important;
+  > *:focus {
+    z-index: 1;
   }
 `;
 
-function determineDirection(direction, flexDirection) {
-  switch (direction) {
-    case Direction.VERTICAL:
-      return "column";
-    case Direction.HORIZONTAL:
-      return "row";
-    default:
-      return flexDirection;
-  }
-}
-
-function determineGutter(gutter, density) {
-  if (gutter !== undefined) {
-    return gutter;
-  } else if (density === Density.COMPACT) {
-    return "2px";
-  } else if (density === Density.COMFORTABLE) {
-    return 1;
-  } else {
-    return 0;
-  }
-}
-
-const Grouper = ({
-  density,
-  direction: dir,
-  flexDirection,
-  gutter: g,
-  ...props
-}) => {
-  const direction = determineDirection(dir, flexDirection);
-  const gutter = determineGutter(g, density);
-  switch (density) {
-    case Density.COMFORTABLE:
-      return (
-        <StyledGrouper flexDirection={direction} gutter={gutter} {...props} />
-      );
-    case Density.COMPACT:
-    default:
-      return (
-        <CompactGrouper flexDirection={direction} gutter={gutter} {...props} />
-      );
-  }
+const Grouper = ({vertical, stretch, gutter, children, ...props}) => {
```

Comment:
Can you explain the thought behind having two parent elements. I'm not quite sure I see the need.

Reply:
`InnerGrouper` uses a “negative margin hack” to grant each child a margin on all sides while eliminating that margin on the outside. Children are given margin on all sides to ensure a gutter not just between siblings, but between rows (or columns) when wrapping. But this ends up indenting everything relative to surrounding elements. The negative margin fixes that. If we used only one parent, this (negative) margin would conflict with any margin props passed in. So we use a wrapper, passing margin props to `OuterGrouper`.